DO App Platform first impressions (#99)
ckcollab/ckcollab: _posts/2020-11-15-digital-ocean-first-impressions.md

---
title: "First impressions: Digital Ocean App Platform"
date: 2020-11-15 00:00:00 +0800
author: eric
---

When I [first learned](https://www.digitalocean.com/blog/introducing-digitalocean-app-platform-reimagining-paas-to-make-it-simpler-for-you-to-build-deploy-and-scale-apps/) about the new Digital Ocean App Platform I was pretty excited to see Heroku have some competition! We use Heroku on many projects and love its simplicity, but hate how quickly it can get expensive.

I attempted to setup one of our client's projects on this new platform to give it a spin, here's my experience...

> <!--more-->

Reviewer: ```suggestion
<!--more-->
I attempted to setup one of our client's projects on this new platform to give it a spin, here's my experience...

```
Might need to move this a little higher. The text doesn't fit well in the preview and causes that row to stretch vertically.
![image](https://user-images.githubusercontent.com/16579931/99318327-567a8f00-281c-11eb-9f33-ec1ac65f926b.png)

Author: I can pull the stretching but it ends up being a bit weird either way... I feel like we should just have 1 column and remove the boxes, can maybe do this in separate PR?

![image](https://user-images.githubusercontent.com/2185159/99319600-a65a5580-281e-11eb-824e-41c2a43dbc01.png)

> # Pros and cons vs Heroku

Reviewer: ```suggestion
# Digital Ocean vs Heroku
```
We can show up in some search results if this title matches. More SEO friendly
![image](https://user-images.githubusercontent.com/16579931/99317280-a6f0ed00-281a-11eb-915b-866f650094d2.png)

Author: good call ty!

<div style="text-align: center;">
    <img src="/assets/images/articles/digital-ocean.png">
</div>

| Pros | Cons |
|------|------|
| Compared to Heroku, there's quite a bit less to learn     | No free database to test against, must pay $7/mo    |
| Scaling up a bit, should be cheaper than Heroku     | Very few add ons     |

> | Fresh, has opportunity to grow with community input     | Many non trivial apps may not be supported because lack of add-ons?     |

Reviewer: `Many non trivial apps may not be supported because of lacking of add-ons?`
Maybe can expand on this more?
You could break this table down into 2 sections. Explaining in more detail these differences. Maybe give some examples as well?

# Cost

The database is minimum $7/mo, no free tier. A little dissappointing, Heroku has a Free tier for Postgres with nice backups and such. It's extremely inviting to have free tiers when tinkering around, has kept me loyal to Heroku for a long time!

Scaling, though, Digital Ocean looks more interesting.

To get a [dyno](https://www.heroku.com/dynos#:~:text=The%20containers%20used%20at%20Heroku,based%20on%20its%20resource%20demands.) with >= 2gb memory on Heroku you need to spend a minimum of **$250/mo**. On Digital Ocean you only need to spend **$20/mo**! 

# Final thoughts

I'm excited to see how this pans out, but what I'd really, _really,_ like to see is some [docker-compose](https://docs.docker.com/compose/) support! It'd be so neat to throw up a server and have new processes managed automatically with nice defaults. Our main concern is business logic, would love to not have to worry so much about configuring yet-another-new-thing :)

> 

Reviewer: You have more posts on Heroku, right? You should add another section where you give some more links people can read. Ideally, we could do this automatically, but doing it manually right now should be fine.
```suggestion
# More to Read
```

Author: added

---
title: "First impressions: Digital Ocean App Platform"
date: 2020-11-15 00:00:00 +0800
author: eric
---

When I [first learned](https://www.digitalocean.com/blog/introducing-digitalocean-app-platform-reimagining-paas-to-make-it-simpler-for-you-to-build-deploy-and-scale-apps/) about the new Digital Ocean App Platform, I was pretty excited to see Heroku have some competition! We use Heroku on many projects and love its simplicity, but hate how quickly it can get expensive.

I attempted to set up one of our client's projects on this new platform to give it a spin. Here is my experience.

<!--more-->

# Digital Ocean vs. Heroku

<div style="text-align: center;">
    <img src="/assets/images/articles/digital-ocean.png">
</div>

| Pros | Cons |
|------|------|
| Compared to Heroku, there's quite a bit less to learn     | No free database to test against, must pay $7/mo    |
| Scaling up a bit should be cheaper than Heroku     | Very few add ons     |

> | Fresh, has the opportunity to grow with community input     | May not support many non-trivial apps because of lacking add-ons     |

Reviewer: ```suggestion
| Fresh, has the opportunity to grow with community input     | May not support many 'complex' or 'enterprise' apps because of lacking add-ons     |
```
Non-trivial isn't very clear here
